src/shared/modules/dbMeta/dbMetaDuck.js

```diff
@@ -339,132 +483,20 @@ export const dbMetaEpic = (some$, store) =>
-                        useCypherThread: shouldUseCypherThread(
-                          store.getState()
-                        ),
-                        ...getBackgroundTxMetadata({
-                          hasServerSupport: canSendTxMetadata(store.getState())
-                        })
-                      }
-                    )
-                    .then(resolve)
-                    .catch(reject)
-                })
-              )
-                .catch(e => {
-                  return Rx.Observable.of(null)
-                })
-                .do(res => {
-                  if (!res) return Rx.Observable.of(null)
-                  const role = res.records[0].get(0)
-                  store.dispatch(update({ role }))
-                  return Rx.Observable.of(null)
-                }),
-              // Database list
-              Rx.Observable.fromPromise(
-                new Promise(async (resolve, reject) => {
-                  const supportsMultiDb = await bolt.hasMultiDbSupport()
-                  if (!supportsMultiDb) {
-                    return resolve(null)
-                  }
-                  bolt
-                    .directTransaction(
-                      'SHOW DATABASES',
-                      {},
-                      {
-                        useCypherThread: shouldUseCypherThread(
-                          store.getState()
-                        ),
-                        ...getBackgroundTxMetadata({
-                          hasServerSupport: canSendTxMetadata(store.getState())
-                        }),
-                        useDb: SYSTEM_DB // System db
-                      }
-                    )
-                    .then(resolve)
-                    .catch(reject)
-                })
-              )
-                .catch(e => {
-                  return Rx.Observable.of(null)
-                })
-                .do(res => {
-                  if (!res) return Rx.Observable.of(null)
-                  const databases = res.records.map(record => ({
-                    ...reduce(
-                      record.keys,
-                      (agg, key) => assign(agg, { [key]: record.get(key) }),
-                      {}
-                    ),
-                    status: record.get('currentStatus')
-                  }))
-
-                  store.dispatch(update({ databases }))
-
-                  // Currently not using a db
-                  if (!getUseDb(store.getState())) {
-                    const defaultDb = databases.filter(db => db.default)
-                    if (defaultDb.length) {
-                      store.dispatch(useDb(defaultDb[0].name))
-                    }
-                  }
-                  return Rx.Observable.of(null)
-                })
+          .mergeMap(() => {
+            return Rx.Observable.forkJoin([
+              getLabelsAndTypes(store),
+              clusterRole(store),
+              databaseList(store)
```

Comment:
Even if one of these errors we'll need to keep the stream going, so I'f like to see some error catching here

Reply:
And if I remember correctly, if one fail the other once are cancelled as well, so I think you need to catch on each one of the inner ones.